### ISSUE TITLE
Inconsistent formatting when running stylua multiple times

### ISSUE THREAD

**Issue by flakolefluk** [JohnnyMorganz/StyLua#397]
given:
```
--[[comment]]
type Doo
--[[comment]]
<
--[[comment]]
T
--[[comment]]
>
--[[comment]]
= 
--[[comment]]
string
--[[comment]]
```

after stylua becomes:
```
--[[comment]]
type Doo--[[comment]]
<
	--[[comment]]
	T	--[[comment]]

>--[[comment]]
 = --[[comment]]
string
--[[comment]]
```
and running it a second time it becomes:
```
--[[comment]]
type Doo--[[comment]]<
	--[[comment]]
	T --[[comment]]
>--[[comment]] =  --[[comment]]string
--[[comment]]
```
running it more times stays the same.

Expected result: 
* results should be the same everytime stylua is executed

Tested on v0.12.4

**Comment by JohnnyMorganz** [comment]
I'm really not sure what the best formatting is to use in this case, I keep playing around with different styles and they don't look great 😅.

Do you possibly have a real-world example to better understand what might make it look good? Or even just a formatting output in mind you think would work

**Comment by matthargett** [comment]
if we reduce it down to this, slightly reduced case:
```lua
--[[opening type comment]]
type Doo<
	T --[[ per-generic argument comment]]
> =
	--[[ opening RHS comment]]
	string --[[ per-RHS comment]]
```

it still takes two runs to format and get
```lua
--[[opening type comment]]
type Doo<
	T --[[ per-generic argument comment]]
> =  --[[ opening RHS comment]]string --[[ per-RHS comment]]
```
It seems like StyLua should retain the trivia on the RHS of the assignment when there's a comment (multi-line or single-line) involved, and that alone may sidestep the reversability issue in this scenario.

**Comment by JohnnyMorganz** [comment]
Some more test cases for this issue available in #430
